src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CachedSystemStoreProvider.cs

```diff
@@ -0,0 +1,276 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed class CachedSystemStoreProvider : IStorePal
+    {
+        private static readonly TimeSpan s_lastWriteRecheckInterval = TimeSpan.FromSeconds(5);
+        private static readonly TimeSpan s_assumeInvalidInterval = TimeSpan.FromMinutes(5);
+        private static readonly Stopwatch s_recheckStopwatch = new Stopwatch();
```

Comment:
This is static because... there's only ever one of these providers?

Reply:
There are two provider instances which are backed by the same atomic static update.  They represent data from a single logical source, so it's just a split projection over a singleton.

Shorter form: Yep :smile:.